File: lowest_unique_number/lowest_unique_number.rb
def lowest_unique_number(line)
  numbers = line.split(' ')
  number_counts = Hash.new(0)

  numbers.each do |number|
    number_counts[number.to_i] += 1
  end

  winning_number = 10

  number_counts.each do |key, value|
    if key < winning_number and value == 1
      winning_number = key
    end
  end

  if winning_number == 10
    return 0
  else
    numbers.each_with_index do |number, index|
      if number.to_i == winning_number
        return index + 1
      end
    end
  end
end

f = File.open(ARGV[0], "r") do |file|
  file.each_line do |line|
    puts lowest_unique_number(line.chomp)
  end
end

File: lowest_unique_number/lowest_unique_number.py
import sys


def main(file_arg):
    with open(file_arg) as f:
        for line in f:
            numbers = line.strip().split(' ')

            numbers = [int(number) for number in numbers]
            sorted_numbers = sorted(numbers)

            unique_numbers = []

            for number in numbers:
                if numbers.count(number) == 1:
                    unique_numbers.append(number)

            if unique_numbers:
                print(numbers.index(min(unique_numbers)) + 1)
            else:
                print(0)


if __name__ == '__main__':
    main(sys.argv[1])